boot/bootutil/src/bootutil_public.c

```diff
@@ -524,26 +523,24 @@ boot_set_pending_multi(int image_index, int permanent)
             rc = boot_write_swap_info(fap, swap_type, 0);
         }
 
-        flash_area_close(fap);
-        return rc;
+        break;
 
     case BOOT_MAGIC_BAD:
         /* The image slot is corrupt.  There is no way to recover, so erase the
          * slot to allow future upgrades.
          */
-        rc = flash_area_open(FLASH_AREA_IMAGE_SECONDARY(image_index), &fap);
-        if (rc != 0) {
-            return BOOT_EFLASH;
-        }
-
         flash_area_erase(fap, 0, fap->fa_size);
-        flash_area_close(fap);
-        return BOOT_EBADIMAGE;
+        rc = BOOT_EBADIMAGE;
+        break;
 
     default:
         assert(0);
-        return BOOT_EBADIMAGE;
+        rc = BOOT_EBADIMAGE;
     }
+
+done:
+    flash_area_close(fap);
```

Comment:
Admittedly, `flash_area_close` doesn't return an error in any current configuration. But we could do something like:
```c
    int rc2 = flash_area_close(fap);
    if (rc == 0) {
        rc = rc2;
    }
```

although, I'm not sure this is really necessary.

Reply:
Unfortunately the original author, of the flash_area_close, did not predict any return value from the close function, so it has void type. I was considering changing that in Zephyr, but dropped the idea.